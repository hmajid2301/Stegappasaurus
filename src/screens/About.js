import PropTypes from 'prop-types';
import React, { Component } from 'react';
import {
  Image,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { Body, Button, Icon, Left, List, ListItem } from 'native-base';
import { StoreReview, WebBrowser } from 'expo';

import Header from '../components/Header';
import { colors, fonts } from '../util/styles';


const logo = require('../assets/images/logo.png');


const icons = [
  {
    title: 'Version 0.1.0',
    icon: {
      name: 'versions',
      type: 'Octicons',
    },
    color: colors.iconOrange,
    url: 'https://github.com/hmajid2301/Stegappasaurus',
  },
  {
    title: 'Fork this Project on GitHub',
    icon: {
      name: 'code-fork',
      type: 'FontAwesome',
    },
    color: colors.iconBlue,
    url: 'https://github.com/hmajid2301/Stegappasaurus',
  },
  {
    title: 'Personal GitHub',
    icon: {
      name: 'github',
      type: 'FontAwesome',
    },
    color: colors.iconBlack,
    url: 'https://github.com/hmajid2301',
  },
  {
    title: 'Personal Website',
    icon: {
      name: 'web',
      type: 'MaterialCommunityIcons',
    },
    color: colors.iconGreen,
    url: 'https://hmajid2301.github.io',
  },
];

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
  },
  textContainer: {
    alignItems: 'center',
    paddingVertical: 20,
  },
  logoImage: {
    height: 50,
    width: 50,
  },
  about: {
    paddingHorizontal: 20,
    fontFamily: fonts.body,
  },
});

export default class About extends Component {
  static navigationOptions = {
    drawerLabel: 'About Us',
    drawerIcon: ({ tintColor }) => (
      <Icon name='info-circle' type='FontAwesome' style={{ color: tintColor }}/>
    ),
  };

  static propTypes = {
    navigation: PropTypes.object.isRequired,
  };

  renderFAQItem = item => (
    <ListItem icon key={item.title}>
      <Left>
        <Button
          onPress={() => WebBrowser.openBrowserAsync(item.url)}
          style={{ backgroundColor: item.color }}
        >
          <Icon {...item.icon} />
        </Button>
      </Left>
      <Body>
        <TouchableOpacity onPress={() => WebBrowser.openBrowserAsync(item.url)}>
          <Text>{item.title}</Text>
        </TouchableOpacity>
      </Body>
    </ListItem>
  );

  render() {
    return (
      <View style={styles.container}>
        <ScrollView>
          <Header color={colors.primary} navigation={this.props.navigation}/>

          <View style={styles.textContainer}>
            <Image source={logo} style={styles.logoImage}/>
            <Text style={styles.about}>
              This project involves implementing steganography algorithms. It allows users
              to hide messages within image files, using these algorithms.
              It was originally developed using the Ionic/Apache Cordova framework.{'\n\n'}

              This app is a rewrite of my third year dissertation project.
              This new app is written using Expo/React Native.
              There are numerous improvements that were made during the rewrite.
              For example the new app has a much better UI/UX and
              has some new features such as sharing the encoded images.
            </Text>
          </View>

          <View>
            <List>
              {
                icons.map(item => (
                  this.renderFAQItem(item)
                ))
              }

              <ListItem icon>
                <Left>
                  <Button
                    onPress={() => StoreReview.requestReview()}
                    style={{ backgroundColor: colors.iconRed }}
                  >
                    <Icon name='rate-review' type='MaterialIcons'/>
                  </Button>
                </Left>
                <Body>
                  <TouchableOpacity onPress={() => StoreReview.requestReview()}>
                    <Text style={styles.body}>Rate the App</Text>
                  </TouchableOpacity>
                </Body>
              </ListItem>

            </List>
          </View>
        </ScrollView>
      </View>
    );
  }
}
