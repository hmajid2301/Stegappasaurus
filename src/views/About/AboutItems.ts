import { IAboutItem } from "~/components/AboutList";
import { colors } from "~/util/styles";

const AboutItems: IAboutItem[] = [
  {
    color: colors.iconOrange,
    icon: {
      name: "versions",
      type: "Octicons"
    },
    title: "Version 0.1.0",
    url: "https://github.com/hmajid2301/Stegappasaurus"
  },
  {
    color: colors.iconBlue,
    icon: {
      name: "code-fork",
      type: "FontAwesome"
    },
    title: "Fork this Project on GitHub",
    url: "https://github.com/hmajid2301/Stegappasaurus"
  },
  {
    color: colors.iconBlack,
    icon: {
      name: "github",
      type: "FontAwesome"
    },
    title: "Personal GitHub",
    url: "https://github.com/hmajid2301"
  },
  {
    color: colors.iconGreen,
    icon: {
      name: "web",
      type: "MaterialCommunityIcons"
    },
    title: "Personal Website",
    url: "https://hmajid2301.github.io"
  },
  {
    color: colors.iconRed,
    function: "store",
    icon: {
      name: "rate-review",
      type: "MaterialIcons"
    },
    title: "Rate the App",
    url: "https://hmajid2301.github.io"
  }
];

export default AboutItems;
