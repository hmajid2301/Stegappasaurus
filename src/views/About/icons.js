import { colors } from '~/util/styles';


const icons = [
  {
    title: 'Version 0.1.0',
    icon: {
      name: 'versions',
      type: 'Octicons',
    },
    color: colors.iconOrange,
    url: 'https://github.com/hmajid2301/Stegappasaurus',
    function: 'browser',
  },
  {
    title: 'Fork this Project on GitHub',
    icon: {
      name: 'code-fork',
      type: 'FontAwesome',
    },
    color: colors.iconBlue,
    url: 'https://github.com/hmajid2301/Stegappasaurus',
    function: 'browser',
  },
  {
    title: 'Personal GitHub',
    icon: {
      name: 'github',
      type: 'FontAwesome',
    },
    color: colors.iconBlack,
    url: 'https://github.com/hmajid2301',
    function: 'browser',
  },
  {
    title: 'Personal Website',
    icon: {
      name: 'web',
      type: 'MaterialCommunityIcons',
    },
    color: colors.iconGreen,
    url: 'https://hmajid2301.github.io',
    function: 'browser',
  },
  {
    title: 'Rate the App',
    icon: {
      name: 'rate-review',
      type: 'MaterialIcons',
    },
    color: colors.iconRed,
    url: 'https://hmajid2301.github.io',
    function: 'store',
  },
];

export default icons;
