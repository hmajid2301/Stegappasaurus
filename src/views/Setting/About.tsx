import React, {useContext} from 'react';
import {Linking, TouchableOpacity, View} from 'react-native';
import {ListItem} from 'react-native-elements';

import Snackbar from '~/actions/Snackbar';
import MarkdownModal from '~/components/MarkdownModal';
import Modal from '~/components/Modal';
import changelog from '~/data/changelog';
import {ThemeContext} from '~/providers/ThemeContext';
import AboutUs from './About/AboutUs';
import Licenses from './About/Licenses';
import {ItemHeaderText, ItemText} from './common';

const About = () => {
  const {background, color} = useContext(ThemeContext).theme;

  return (
    <View>
      <ListItem
        containerStyle={{
          backgroundColor: background,
        }}
        titleStyle={{
          paddingBottom: 5,
          paddingTop: 20,
        }}
        title={<ItemHeaderText>About</ItemHeaderText>}
      />

      <ListItem
        containerStyle={{
          backgroundColor: background,
        }}
        title={
          <Modal name="About Us">
            <AboutUs />
          </Modal>
        }
        topDivider={true}
      />

      <ListItem
        containerStyle={{
          backgroundColor: background,
        }}
        title={
          <Modal name="Licenses">
            <Licenses />
          </Modal>
        }
        topDivider={true}
        bottomDivider={true}
      />

      <ListItem
        containerStyle={{
          backgroundColor: background,
        }}
        title={<MarkdownModal name="Changelog">{changelog}</MarkdownModal>}
        topDivider={true}
        bottomDivider={true}
      />

      <ListItem
        containerStyle={{
          backgroundColor: background,
        }}
        title={
          <View>
            <TouchableOpacity onPress={sendEmail}>
              <ItemText color={color}>Email</ItemText>
            </TouchableOpacity>
          </View>
        }
        subtitle={<ItemText color="grey">me@haseebmajid.dev</ItemText>}
        bottomDivider={true}
      />
    </View>
  );
};

const sendEmail = async () => {
  try {
    await Linking.openURL('mailto:me@haseebmajid.dev?subject=Stegappasaurus');
  } catch {
    Snackbar.show({
      text: 'Failed to open an email app.',
    });
  }
};

export default About;
