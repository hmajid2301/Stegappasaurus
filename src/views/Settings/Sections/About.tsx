import * as React from "react";
import { Linking, Text, TouchableOpacity, View } from "react-native";
import { ListItem } from "react-native-elements";

import Snackbar from "~/actions/Snackbar";
import MarkdownModal from "~/components/MarkdownModal";
import changelog from "~/data/changelog";
import { ThemeColors } from "~/modules/types";
import styles from "./styles";

interface IProps {
  background: ThemeColors;
  color: ThemeColors;
}

const About = ({ background, color }: IProps) => (
  <View>
    <ListItem
      containerStyle={{
        backgroundColor: background
      }}
      titleStyle={styles.itemHeader}
      title={<Text style={styles.itemHeaderText}>About</Text>}
    />
    <ListItem
      containerStyle={{
        backgroundColor: background
      }}
      title={
        <MarkdownModal background={background} color={color} name="Changelog">
          {changelog}
        </MarkdownModal>
      }
      topDivider={true}
      bottomDivider={true}
    />

    <ListItem
      containerStyle={{
        backgroundColor: background
      }}
      title={
        <View>
          <TouchableOpacity onPress={sendEmail}>
            <Text style={[styles.itemText, { color }]}>Email</Text>
            <Text style={[styles.itemText, styles.itemTextUnder]}>
              me@haseebmajid.dev
            </Text>
          </TouchableOpacity>
        </View>
      }
      bottomDivider={true}
    />
  </View>
);

const sendEmail = async () => {
  try {
    await Linking.openURL("mailto:me@haseebmajid.dev?subject=Stegappasaurus");
  } catch {
    Snackbar.show({
      text: "No email client installed on your device, failed to send email."
    });
  }
};

export default About;
