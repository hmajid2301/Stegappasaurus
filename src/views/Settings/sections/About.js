import {
  Body,
  List,
  ListItem,
} from 'native-base';
import PropTypes from 'prop-types';
import React from 'react';
import { Text, TouchableOpacity } from 'react-native';

import { Changelog } from './Modals';

import styles from './styles';


const About = ({ theme }) => (
  <List>

    <ListItem itemHeader style={styles.itemHeader}>
      <Text style={styles.itemHeaderText}>About</Text>
    </ListItem>

    <ListItem noIndent>
      <Changelog />
    </ListItem>

    <ListItem noIndent>
      <Body>
        <TouchableOpacity onPress={() => this.sendEmail()}>
          <Text style={[styles.itemText, { color: theme.color }]}>Email</Text>
          <Text style={[styles.itemText, styles.itemTextUnder]}>me@haseebmajid.com</Text>
        </TouchableOpacity>
      </Body>
    </ListItem>
  </List>
);

About.propTypes = {
  theme: PropTypes.object.isRequired,
};

export default About;
