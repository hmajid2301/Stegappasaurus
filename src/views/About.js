import { Icon } from 'native-base';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { ScrollView, Text, View } from 'react-native';

import CustomHeader from '~/components/CustomHeader';
import AboutList from '~/views/About/AboutList';
import icons from '~/views/About/icons';
import styles from '~/views/About/styles';
import { colors } from '~/util/styles';


export default class About extends Component {
  static navigationOptions = {
    drawerLabel: 'About Us',
    drawerIcon: ({ tintColor }) => (
      <Icon name='info' type='Feather' style={{ color: tintColor }}/>
    ),
  };

  static propTypes = {
    navigation: PropTypes.object.isRequired,
    screenProps: PropTypes.shape({
      theme: PropTypes.shape({
        isDark: PropTypes.bool.isRequired,
        background: PropTypes.string.isRequired,
        color: PropTypes.string.isRequired,
      }),
    }),
  };

  render() {
    const { theme } = this.props.screenProps;

    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <ScrollView>
          <CustomHeader
            primaryColor={colors.primary}
            navigation={this.props.navigation}
            theme={theme}
          />

          <View style={styles.textContainer}>
            <Text style={[styles.about, { color: theme.color }]}>
              This project involves implementing steganography algorithms. It allows users
              to hide messages within image files, using these algorithms.
              It was originally developed using the Ionic/Apache Cordova framework.{'\n\n'}

              This app is a rewrite of my third year dissertation project.
              This new app is written using Expo/React Native.
              There are numerous improvements that were made during the rewrite.
              For example the new app has a much better UI/UX and
              has some new features such as sharing the encoded images.
            </Text>
          </View>

          <View>
            <AboutList icons={icons} color={theme.color}></AboutList>
          </View>
        </ScrollView>
      </View>
    );
  }
}
