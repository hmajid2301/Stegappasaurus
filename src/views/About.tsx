import { Icon } from "native-base";
import React, { Component } from "react";
import { ScrollView, Text, View } from "react-native";
import { NavigationScreenProp } from "react-navigation";

import CustomHeader from "~/components/CustomHeader";
import { colors } from "~/util/styles";
import AboutList from "~/views/About/AboutList";
import icons from "~/views/About/icons";
import styles from "~/views/About/styles";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  screenProps: {
    theme: {
      background: string;
      color: string;
      isDark: boolean;
    };
  };
}

interface IColor {
  tintColor: string;
}

export default class About extends Component<IProps, {}> {
  public static navigationOptions = {
    drawerIcon: ({ tintColor }: IColor) => (
      <Icon name="info" type="Feather" style={{ color: tintColor }} />
    ),
    drawerLabel: "About Us"
  };

  public render() {
    const { theme } = this.props.screenProps;

    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <ScrollView>
          <CustomHeader
            primaryColor={colors.primary}
            navigation={this.props.navigation}
            theme={theme}
          />

          <View style={styles.textContainer}>
            <Text style={[styles.about, { color: theme.color }]}>
              This project involves implementing steganography algorithms. It
              allows users to hide messages within image files, using these
              algorithms. It was originally developed using the Ionic/Apache
              Cordova framework.{"\n\n"}
              This app is a rewrite of my third year dissertation project. This
              new app is written using Expo/React Native. There are numerous
              improvements that were made during the rewrite. For example the
              new app has a much better UI/UX and has some new features such as
              sharing the encoded images.
            </Text>
          </View>

          <View>
            <AboutList icons={icons} color={theme.color} />
          </View>
        </ScrollView>
      </View>
    );
  }
}
