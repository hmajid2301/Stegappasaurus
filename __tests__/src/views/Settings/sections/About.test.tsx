import { mount, shallow } from "enzyme";
import React from "react";

import About from "~/views/Settings/sections/About";

describe("About: Match Snapshots", () => {
  test("1", () => {
    const component = shallow(<About color="#17212D" background="#FFF" />);
    expect(component).toMatchSnapshot();
  });

  test("2", () => {
    const component = shallow(<About background="#17212D" color="#FFF" />);
    expect(component).toMatchSnapshot();
  });
  test("3", () => {
    const component = mount(<About background="#17212D" color="#FFF" />);
    expect(
      component
        .find("ListItem")
        .at(1)
        .props().title
    ).toMatchSnapshot();
  });
});

describe("About: Props", () => {
  test("OnPress", async () => {
    const component = mount(<About background="#17212D" color="#FFF" />);
    const prop = component
      .find("TouchableOpacity")
      .last()
      .props() as any;
    prop.onPress();
    expect(prop).toBeTruthy();
  });
});
