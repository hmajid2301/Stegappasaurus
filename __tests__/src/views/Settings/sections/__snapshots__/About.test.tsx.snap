// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About: Match Snapshots 1 1`] = `
<View>
  <Themed.ListItem
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    title={
      <Text
        style={
          Object {
            "color": "#009CFF",
            "fontFamily": "Roboto",
            "fontSize": 12,
          }
        }
      >
        About
      </Text>
    }
    titleStyle={
      Object {
        "paddingBottom": 5,
        "paddingTop": 20,
      }
    }
  />
  <Themed.ListItem
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    title={
      <MyModal
        background="#FFF"
        color="#17212D"
        name="About Us"
      >
        <AboutUs
          background="#FFF"
          color="#17212D"
        />
      </MyModal>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    title={
      <MyModal
        background="#FFF"
        color="#17212D"
        name="Licenses"
      >
        <Licenses
          background="#FFF"
          color="#17212D"
        />
      </MyModal>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    title={
      <MarkdownModal
        background="#FFF"
        color="#17212D"
        name="Changelog"
      >
        
# Changelog

## 1.0.0
### What's New

* Initial release of the app
      </MarkdownModal>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#FFF",
      }
    }
    title={
      <View>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Roboto-Light",
                },
                Object {
                  "color": "#17212D",
                },
              ]
            }
          >
            Email
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Roboto-Light",
                },
                Object {
                  "color": "grey",
                },
              ]
            }
          >
            me@haseebmajid.dev
          </Text>
        </TouchableOpacity>
      </View>
    }
  />
</View>
`;

exports[`About: Match Snapshots 2 1`] = `
<View>
  <Themed.ListItem
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    title={
      <Text
        style={
          Object {
            "color": "#009CFF",
            "fontFamily": "Roboto",
            "fontSize": 12,
          }
        }
      >
        About
      </Text>
    }
    titleStyle={
      Object {
        "paddingBottom": 5,
        "paddingTop": 20,
      }
    }
  />
  <Themed.ListItem
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    title={
      <MyModal
        background="#17212D"
        color="#FFF"
        name="About Us"
      >
        <AboutUs
          background="#17212D"
          color="#FFF"
        />
      </MyModal>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    title={
      <MyModal
        background="#17212D"
        color="#FFF"
        name="Licenses"
      >
        <Licenses
          background="#17212D"
          color="#FFF"
        />
      </MyModal>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    title={
      <MarkdownModal
        background="#17212D"
        color="#FFF"
        name="Changelog"
      >
        
# Changelog

## 1.0.0
### What's New

* Initial release of the app
      </MarkdownModal>
    }
    topDivider={true}
  />
  <Themed.ListItem
    bottomDivider={true}
    containerStyle={
      Object {
        "backgroundColor": "#17212D",
      }
    }
    title={
      <View>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Roboto-Light",
                },
                Object {
                  "color": "#FFF",
                },
              ]
            }
          >
            Email
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Roboto-Light",
                },
                Object {
                  "color": "grey",
                },
              ]
            }
          >
            me@haseebmajid.dev
          </Text>
        </TouchableOpacity>
      </View>
    }
  />
</View>
`;

exports[`About: Match Snapshots 3 1`] = `
<MyModal
  background="#17212D"
  color="#FFF"
  name="About Us"
>
  <AboutUs
    background="#17212D"
    color="#FFF"
  />
</MyModal>
`;
